### PR TITLE
 updated Crunchy Bridge operator to retrieve instance connection details for DBaaS

### PR DESCRIPTION
**Pull request by redhatHameed**
This Pr change will update CrunchyBridgeConnection CR to sett connection info in status as shown ->
```
spec:
  instanceID: na7wuu36f5f67oaxm75ngjl5pe
  inventory:
    name: crunchybridgeinventory-sample
status:
  conditions:
    - lastTransitionTime: '2021-06-18T16:37:22Z'
      message: Successfully reterieved the connection detail
      reason: ReadyForBinding
      status: 'True'
      type: ReadyForBinding
  connectionInfo:
    instanceId: na7wuu36f5f67oaxm75ngjl5pe
  connectionString: >-
    postgres://@p<url>:<>/postgres
  credentialsRef:
    name: crunchy-bridge-db-user-s4vgt

```

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -31,3 +31,28 @@
 - **Next edit the [catalog-source.yaml](config/samples/catalog-source.yaml) template to indicate your new Quay.io org image**  
 - `make release catalog-update`
 - search the Crunchy Bridge Operator in OperatorHub, click on install.
+
+## DBaaS Integration
+
+Follow the steps below to enable integration with DBaaS.
```
Comment (bmozaffa) [inline README.md:37]: This needs some clarification. The 3 steps below are not needed to integrate with DBaaS. If anything, they're needed to unit test the functionality **without** integrating with the DBaaS operator

Comment (redhatHameed) [reply README.md]: > This needs some clarification. The 3 steps below are not needed to integrate with DBaaS. If anything, they're needed to unit test the functionality **without** integrating with the DBaaS operator

Thanks, @bmozaffa, updated the description.

File: controllers/dbaas.redhat.com/crunchybridgeconnection_controller.go [hunk 2]
```diff
@@ -47,10 +54,60 @@ type CrunchyBridgeConnectionReconciler struct {
...
+			// CR deleted since request queued, child objects getting GC'd, no requeue
+			logger.Info("inventory resource not found, has been deleted")
+			return ctrl.Result{}, err
+		}
+		logger.Error(err, "Error fetching CrunchyBridgeConnection for reconcile")
+		return ctrl.Result{}, err
+	}
+	instance := getInstance(&inventory, connection.Spec.InstanceID)
+	if instance == nil {
+		return ctrl.Result{}, err
```
Comment (wilybrace) [inline controllers/dbaas.redhat.com/crunchybridgeconnection_controller.go:82]: err has not been updated for this block, is this intentional or should it be emitting a newer, custom error?

Comment (redhatHameed) [reply controllers/dbaas.redhat.com/crunchybridgeconnection_controller.go:84]: updated, now returning the error from getInstance.

**Pull request merged**